ansible/roles/operator-pipeline/templates/openshift/tasks/upload-signature.yml

```diff
@@ -0,0 +1,73 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: upload-signature
+spec:
+  params:
+    - name: docker_reference
+    - name: manifest_digest
+    - name: repository
+    - name: sig_key_id
+      description: The signing key id that the content was signed with.
+    - name: signature_data
+      description: The signature to upload to Pyxis signature.
+    - name: pipeline_image
+    - name: pyxis_ssl_secret_name
+      description: Kubernetes secret name that contains the Pyxis SSL files.
+    - name: pyxis_ssl_cert_secret_key
+      description: The key within the Kubernetes secret that contains the Pyxis SSL cert.
+    - name: pyxis_ssl_key_secret_key
+      description: The key within the Kubernetes secret that contains the Pyxis SSL key.
+    - name: pyxis_url
+      description: Pyxis instance to upload the signature to.
+      default: https://pyxis.engineering.redhat.com
+    - name: verify_signature
+      description: Whether to verify that the signature data is signed with the right key.
+      default: "true"
+  volumes:
+    - name: pyxis-ssl-volume
+      secret:
+        secretName: "$(params.pyxis_ssl_secret_name)"
+        optional: false
+  steps:
+    - name: verify-signature
+      image: "$(params.pipeline_image)"
+      script: |
+        #! /usr/bin/env bash
+        set -xe
+
+        if [[ "$(params.verify_signature)" == "true" ]]; then
+          echo "Verifying signature before upload"
+          echo $(params.signature_data) | base64 --decode > decoded_signed_claim
+
+          # import containerisvsign pub key
```

Comment:
There might be a better way to import the pub key, but trying to bake it into the container image didn't work (even putting it after the user change). If you can think of a better way please let me know.

Reply:
One option is to download the key during the image build and set an environment variable to its path. Continue to run the `gpg` commands in this task but reference the env var. If it's possible, parametrize the env variable name in this task too.

Wdyt? Do you think that'll work?

Reply:
Yup, that works! Updated, thanks for the input.